src/warc2zim/items.py

```diff
@@ -8,50 +8,58 @@
 """
 
 import logging
-import re
-
 import pkg_resources
 from libzim.writer import Hint
 from zimscraperlib.types import get_mime_for_name
 from zimscraperlib.zim.items import StaticItem
 
+from warcio.recordloader import ArcWarcRecord
+
 from warc2zim.utils import get_record_url, get_record_mime_type
-from warc2zim.content_rewriting import HtmlRewriter, CSSRewriter
+from warc2zim.url_rewriting import ArticleUrlRewriter
+from warc2zim.content_rewriting import HtmlRewriter, CSSRewriter, JSRewriter
 
 # Shared logger
 logger = logging.getLogger("warc2zim.items")
 
-# external sw.js filename
-SW_JS = "sw.js"
-
-HEAD_INS = re.compile(b"(<head>)", re.I)
-CSS_INS = re.compile(b"(</head>)", re.I)
-
 
 class WARCPayloadItem(StaticItem):
     """WARCPayloadItem used to store the WARC payload
     Usually stored under A namespace
     """
 
-    def __init__(self, path, record, head_insert, css_insert):
+    def __init__(
+        self, path: str, record: ArcWarcRecord, head_template: str, css_insert: str
+    ):
         super().__init__()
         self.record = record
         self.path = path
         self.mimetype = get_record_mime_type(record)
         self.title = ""
 
+        url_rewriter = ArticleUrlRewriter(self.path)
         if hasattr(self.record, "buffered_stream"):
             self.record.buffered_stream.seek(0)
             self.content = self.record.buffered_stream.read()
         else:
             self.content = self.record.content_stream().read()
 
+        should_rewrite = getattr(record, "method", "GET") != "POST"
+        if not should_rewrite:
+            return
+
         if self.mimetype.startswith("text/html"):
+            wombat_path = url_rewriter.from_normalized("_zim_static/wombat.js")
+            head_insert = head_template.render(
+                path=path, wombat_path=wombat_path, url=get_record_url(record)
```

Comment:
Ain't it missing `orig_scheme` and `orig_host` (and maybe others)?

Reply:
Yes indeed. I missed a commit during the cleanup of the git history.
See the two last commits.